torchdata/datapipes/iter/util/dataframemaker.py

```diff
@@ -44,3 +46,48 @@ def __new__(
         batch_dp = source_dp.batch(dataframe_size)
         df_dp = batch_dp.map(partial(torcharrow.DataFrame, dtype=dtype, columns=columns, device=device))
         return df_dp
+
+
+@functional_datapipe("load_parquet_as_df")
+class ParquetDFLoaderIterDataPipe(IterDataPipe):  # IterDataPipe[torcharrow.IDataFrame[T_co]]
+    r"""
+    Iterable DataPipe that takes in paths to Parquet files and return a TorchArrow DataFrame for each row group
+    within a Parquet file.
+
+    Args:
+        source_dp: source DataPipe containing paths to the Parquet files
+        columns: List of str that specifies the column names of the DataFrame
+        use_threads: if True, Parquet reader will perform multi-threaded column reads
+        dtype: specify the TorchArrow dtype for the DataFrame
+        device: specify the device on which the DataFrame will be stored
+    """
+
+    def __init__(
+        self,
+        source_dp: IterDataPipe[str],
+        dtype=None,  # Optional[torcharrow.dtypes.DType]
+        columns: Optional[List[str]] = None,
+        device: str = "",
+        use_threads: bool = False,
+    ):
+        if torcharrow is None:
+            raise ImportError(
+                "The library 'torcharrow' is necessary for this DataPipe but it is not available."
+                "Please visit https://github.com/facebookresearch/torcharrow/ to install it."
+            )
+        if parquet is None:
+            raise ImportError("The library 'parquet' is necessary for this DataPipe but it is not available.")
+        self.source_dp = source_dp
+        self.columns = columns
+        self.use_threads = use_threads
```

Comment:
Does it makes it non-deterministic?

Reply:
Are you asking if we will get the same DataFrame every time (with the same ordering of rows) when multi-threading?

I believe PyArrow's implementation of read from Parquet file is deterministic.

Reply:
SGTM